src/py/flwr/server/superlink/state/in_memory_state.py

```diff
@@ -287,6 +297,27 @@ def get_client_public_keys(self) -> Set[bytes]:
         """Retrieve all currently stored `client_public_keys` as a set."""
         return self.client_public_keys
 
+    def get_node_id(self, client_public_key: bytes) -> int:
+        """Retrieve stored `node_id` filtered by `client_public_keys`."""
+        print("nodes: ", self.node_ids)
+        print("pairs: ", self.public_key_node_id_pairs)
+        return self.public_key_node_id_pairs[client_public_key]
+
+    def store_node_id_client_public_key_pair(
+        self, client_public_key: bytes, node_id: int
+    ) -> None:
+        """Store `node_id` and `client_public_keys` as pairs."""
+        self.public_key_node_id_pairs[client_public_key] = node_id
+        print(self.public_key_node_id_pairs)
```

Comment:
Never use `print` in framework code. Use `log` instead.

Reply:
Removed.